packages/serverless/src/domain/use-case/database.ts

```diff
@@ -44,16 +46,21 @@ export async function getDatabases(
   pagination?: Pagination
 ): Promise<Database[]> {
   try {
-    const databasesInfo = await DatabaseEngine.getInstance()
-      .client.db()
-      .admin()
-      .listDatabases();
+    const dbEngine = DatabaseEngine.getInstance();
+    const databasesInfo = await dbEngine.client.db().admin().listDatabases();
 
-    const databaseInfoMap: { [name: string]: { sizeOnDisk: number } } = {};
+    if (!databasesInfo?.databases?.length) {
+      return [];
+    }
 
-    databasesInfo.databases.forEach((dbInfo) => {
-      databaseInfoMap[dbInfo.name] = { sizeOnDisk: dbInfo.sizeOnDisk || 0 };
-    });
+    const databaseInfoMap: Record<string, { sizeOnDisk: number }> =
+      databasesInfo.databases.reduce(
+        (acc, dbInfo) => {
+          acc[dbInfo.name] = { sizeOnDisk: dbInfo.sizeOnDisk || 0 };
+          return acc;
+        },
+        {} as Record<string, { sizeOnDisk: number }>
+      );
```

Comment:
```suggestion
    const databaseInfoMap: Record<string, { sizeOnDisk: number }> = 
  databasesInfo.databases.reduce<Record<string, { sizeOnDisk: number }>>(
    (acc, dbInfo) => ({
      ...acc,
      [dbInfo.name]: { sizeOnDisk: dbInfo.sizeOnDisk || 0 }
    }),
    {}
  );
```

Reply:
This provide you with more readability. But consider if the databases record too big. You should use forEach or regular for to get the best perfomance